pkg/pool-weighted/test/ManagedPool.test.ts

```diff
@@ -658,7 +657,114 @@ describe('ManagedPool', function () {
       });
     });
 
-    describe('management fees', () => {
+    describe('protocol fee cache update', () => {
+      let vault: Vault;
+      const swapFeePercentage = fp(0.02);
+      const managementSwapFeePercentage = fp(0.8);
+
+      sharedBeforeEach('deploy pool', async () => {
+        vault = await Vault.create();
+
+        const params = {
+          tokens: poolTokens,
+          weights: poolWeights,
+          owner: owner.address,
+          poolType: WeightedPoolType.MANAGED_POOL,
+          swapEnabledOnStart: true,
+          vault,
+          swapFeePercentage,
+          managementSwapFeePercentage,
+        };
+        pool = await WeightedPool.create(params);
+      });
+
+      it('emits event when protocol fee cache is updated', async () => {
+        const receipt = await pool.instance.updateCachedProtocolSwapFeePercentage();
+
+        // Real Vault will return a zero protocol fee
+        expectEvent.inReceipt(await receipt.wait(), 'ProtocolSwapFeeCacheUpdated', {
+          protocolSwapFeePercentage: 0,
+        });
+      });
+    });
+
+    describe.skip('BPT protocol fees', () => {
```

Comment:
Are the skipped tests a way to signal that they need writing?

Reply:
There were open questions at the time about whether we were going to pay protocol fees on joinSwaps, etc., and I wanted to get clarity on that before fleshing out all the tests.

I will look at these again; some could probably be written now.